schema-loader/src/main/java/com/scalar/db/schemaloader/SchemaOperator.java

```diff
@@ -25,23 +28,33 @@
 public class SchemaOperator implements AutoCloseable {
   private static final Logger logger = LoggerFactory.getLogger(SchemaOperator.class);
 
-  private final DistributedStorageAdmin storageAdmin;
-  private final DistributedTransactionAdmin transactionAdmin;
+  private final Supplier<DistributedStorageAdmin> storageAdmin;
+  private final AtomicBoolean storageAdminLoaded = new AtomicBoolean();
+  private final Supplier<DistributedTransactionAdmin> transactionAdmin;
+  private final AtomicBoolean transactionAdminLoaded = new AtomicBoolean();
   private final TableMetadataAlterationProcessor alterationProcessor;
 
   public SchemaOperator(Path propertiesFilePath) throws IOException {
-    StorageFactory storageFactory = StorageFactory.create(propertiesFilePath);
-    storageAdmin = storageFactory.getStorageAdmin();
-    TransactionFactory transactionFactory = TransactionFactory.create(propertiesFilePath);
-    transactionAdmin = transactionFactory.getTransactionAdmin();
-    alterationProcessor = new TableMetadataAlterationProcessor();
+    this(StorageFactory.create(propertiesFilePath), TransactionFactory.create(propertiesFilePath));
   }
 
   public SchemaOperator(Properties properties) {
-    StorageFactory storageFactory = StorageFactory.create(properties);
-    storageAdmin = storageFactory.getStorageAdmin();
-    TransactionFactory transactionFactory = TransactionFactory.create(properties);
-    transactionAdmin = transactionFactory.getTransactionAdmin();
+    this(StorageFactory.create(properties), TransactionFactory.create(properties));
+  }
+
+  private SchemaOperator(StorageFactory storageFactory, TransactionFactory transactionFactory) {
+    storageAdmin =
+        Suppliers.memoize(
+            () -> {
+              storageAdminLoaded.set(true);
+              return storageFactory.getStorageAdmin();
```

Comment:
[minor] Maybe we can write a test to confirm `storageFactory.getStorageAdmin()` or `transactionFactory.getTransactionAdmin()` isn't called.

Reply:
Correct. I'll do that. Thanks.

Reply:
@komamitsu Added unit tests in 7fee183d7e48234446e27350059fffff57edccfb. Please take a look again when you have time!